contributor-growth/drafts/contributor-growth-framework/1-Motivating.md

```diff
@@ -0,0 +1,107 @@
+   * Clear PR submission requirements (e.g. Contributor License Agreement, Digital Certificate of Origin)
+   * Submitting the PR (and be sure to set the expectations right regarding wait time, but more to that below)
+
+### Documentation
+
+Good docs can be a motivator. It makes it easier to use the project but also shows that the project runs efficiently, that there is a group of responsible engineers behind the project, and that it will likely be a well-organized, well-specified experience. 
+Specify what contributors should expect when they open their first PR. Point to other docs that explain what a PR and branch is and what the workflow looks like. Not everyone may be familiar with it. Consider how your workflow differs from standard workflow concepts and emphasizing those differences through documentation. And make sure contributors at all levels have the tools they need.
+While we all know docs are important, the challenge is often prioritization. If growing contributors is a priority, focus on the first time experience.  
+
+Writing tests can also be a contribution barrier. How does your project expect their tests? Document it and be specific.  
+Also, be clear about where to ask questions if they can't find the answer in the docs. 
+While you may have onboarding docs, they may be outdated or incomplete (after all, when was the last time you needed them?). Leverage new contributors'  "beginner's mind" by encouraging them to improve and/or fix them or at least interview them after they submit their first PR. What's weird, broken, or unclear? If something isn't documented, encourage them to go ahead and add it.
+
+### Minimize Steps: Developer Environment Experience 
+
+The first setup experience is key. Make sure that the development requirements are clear. What languages does the project use? Where can a contributor find the tooling that they need to install to be successful during development? Leverage any tool that reduces steps or makes the entire process smoother.  
+
+Use scripts to install tooling or tests and be clear about what is being installed. Ideally, all your contributor needs is one command to set up the environment, maybe two to run tests. 
+
+If your project has external dependencies, Docker Compose is a great tool.  It allows you to put a database into a Docker Compose file, and let the contributor run docker-compose up. 
+And make sure everything is well documented in your docs and/or video tutorials. No matter how trivial a step is, if it's part of the process, document it. Remember that friction is dangerous, so always ask yourself "can I make this easier?"  
+
+Examples of tools that help minimize PR submission process
+Tool | What it does
+---|---
+Tilt | [Automate a lot of otherwise manual steps, significantly improving the contributor experience. ] >> needs proper description
+Docker compose | To put dependencies in a Docker Compose file
+docker | To run docker containers locally
```

Comment:
A lot of projects use docker not just to run their code, but to create a consistent development environment for building it. These docker utility containers have all the tools you need, on the right version, etc so that anyone with Docker can build the code.

Reply:
added